src/redis-cli.c

```diff
@@ -3593,6 +3593,7 @@ static int evalMode(int argc, char **argv) {
         /* Call it */
         int eval_ldb = config.eval_ldb; /* Save it, may be reverted. */
         retval = issueCommand(argc+3-got_comma, argv2);
+        free(argv2);
```

Comment:
```suggestion
        for (j = 0; i < argc+3-got_comma; i++) sdsfree(argv2[i]);
        free(argv2);
```

Reply:
not yet tested it, please check it again.

Reply:
> 

The allocation of the dynamic memory area pointed to by argv2 is at line 3580: argv2 = zmalloc(sizeof(sds)*(argc+3)); Therefore, should all dynamic memory areas between 0 and argc+3 be freed (rather than 0 to argc+3-got_comma), because the dynamic memory area pointed to by argv2 will no longer be used after this point.

Reply:
please note the fowllowing code.
if `got_comma` is 1, mean argv2 will be populated one less sds.
i guess that you confuse between argv2 and the individual sds in argv2.
```
            if (!got_comma && argv[j][0] == ',' && argv[j][1] == 0) {
                got_comma = 1;
                continue;
            }
```

Reply:
> please note the fowllowing code.
> if `got_comma` is 1, mean argv2 will be populated one less sds.
> i guess that you confuse between argv2 and the individual sds in argv2.

Thank you for your patient explanation! It was my oversight earlier.